### PR TITLE
ejected app

### LINKED ISSUE #22: Eject Application to Bare Workflow
# Goal

The app should be ejected from Expo so that we can use React Native libraries

# Plan

1. Refer to Expo docs for ejecting
1. Ensure we are still able to build and launch the app after ejecting

### PR DESCRIPTION
**Pull request by C-Begley**
# Details

- Ejected app, added any expanded code to be included. Double checked react ignore to ensure that all build files are ignored

# Screenshots/GIF (if appropriate)
Screenshot from 2021-02-16 09-43-59
Screenshot from 2021-02-16 09-44-08

# Checklist

## Code
- [ ] I have added some tests
- [ ] I have run the tests locally to ensure they all pass

## Documentation
- [x] I have updated documentation appropriately (GitHub Pages)
- [x] I have tested the updated GitHub Pages locally to ensure the changes to documentation render correctly (see this [Guidance](https://ese-peasy.github.io/PosturePerfection/guidance.html) page)

## Accessibility
- [x] Tested Accessibility on **Android**
  - [ ] Any added fields can be read aloud by the screen reader
- [x] Tested Accessibility on **iOS**
  - [ ] Any added fields can be read aloud by the screen reader

Closes #22

### CODE REVIEW

**Comment by C-Begley** [comment]
Check your device is connected to your laptop:
`adb devices`
Run the react sever
`npx react-native start`
Build and load app
`npm run android`

**Comment by C-Begley** [comment]
Down to 27,000 from 140,000 additions, so that has to be an improvement

File: app/app.json [hunk 1]
```diff
@@ -19,9 +19,13 @@
       "**/*"
     ],
     "ios": {
-      "supportsTablet": true
+      "supportsTablet": true,
+      "bundleIdentifier": "com.esepeasy.narratemyway",
+      "buildNumber": "1.0.0"
```
Comment (miklasr) [inline app/app.json:24]: Should this match the version number?

File: app/app.json [hunk 2]
```diff
@@ -19,9 +19,13 @@
...
     },
     "android": {
+      "package": "com.esepeasy.narratemyway",
+      "versionCode": 1,
```
Comment (miklasr) [inline app/app.json:28]: Should this match the version number?

**Review by miklasr — COMMENTED** [review]
I still need to setup things to test if it runs on my phone, but I have a small question about version numbers in the code. Just want to double check that the fields I highlighted have the correct number. I could be wrong and it may be something to do with a compiler version to use for building or something else.

File: app/app.json (hunk 2 above)
Comment (C-Begley) [reply app/app.json:28]: So android has version numbers and version codes, and version codes have to start at 1 and are integers

**Comment by C-Begley** [comment]
`cd app`
`npm install`
`npm fund`
`npm install -g expo-cli`
`npm fund`
`npm audit fix`
(Don't know how to handle these vulnerabilities, can perhaps force)
Follow https://reactnative.dev/docs/environment-setup
`npx react-native start`
`npm run android`

**Comment by miklasr** [comment]
> `cd app`
> `npm install`
> `npm fund`
> `npm install -g expo-cli`
> `npm fund`
> `npm audit fix`
> (Don't know how to handle these vulnerabilities, can perhaps force)
> Follow https://reactnative.dev/docs/environment-setup
> `npx react-native start`
> `npm run android`

I don't think we should run `npm fix`, because it can break certain things if versions are updated automatically.

**Comment by C-Begley** [comment]
> > `cd app`
> > `npm install`
> > `npm fund`
> > `npm install -g expo-cli`
> > `npm fund`
> > `npm audit fix`
> > (Don't know how to handle these vulnerabilities, can perhaps force)
> > Follow https://reactnative.dev/docs/environment-setup
> > `npx react-native start`
> > `npm run android`
> 
> I don't think we should run `npm fix`, because it can break certain things if versions are updated automatically.

I think audit fix is okay? We just shouldn't maybe force

**Comment by miklasr** [comment]
Should all the files in the build directory definitely be pushed? I know you were having problems without it, but it sounds like a directory that shouldn't be committed.

**Comment by Ashwin-MJ** [comment]
I think we want to add `app/android/build/`, `app/android/app/build/` and `app/ios/Pods/` (generated via XCode project) to the `.gitignore`

**Comment by miklasr** [comment]
For reference, I did the following to get things running on my end:

**Set up android environment on Ubuntu**
Follow instructions [here](https://reactnative.dev/docs/environment-setup).

I installed android studio via `snap`:
```sh
sudo snap install android-studio --classic
```
To get the correct Java version:
```sh
sudo apt install default-jdk
```
**Running the application**

```sh
cd app
npm install
npm install -g expo-cli
npx react-native start
```
In another terminal:
```sh
npm run android
```

File: .gitignore [hunk 3]
```diff
@@ -85,7 +85,7 @@ node_modules
 
 # Root dir shouldn't have Xcode project
 /*.xcodeproj
-
+:wq
```
Comment (Ashwin-MJ) [inline .gitignore]: Trouble getting out of Vim? 😉

**Review by Ashwin-MJ — DISMISSED** [review]
Looks good, I'm happy to merge in unless anyone else spots any issues

File: docs/development/installation.md [hunk 4]
```diff
@@ -30,12 +30,51 @@ Follow the instructions [here](https://reactnative.dev/docs/environment-setup) f
    cd app
    npm install
    ```
+1. Install various dependencies for your OS system and android/Iosrun  [here](https://reactnative.dev/docs/environment-setup)
+
+### Linux android setup
+Steps take from [here](https://reactnative.dev/docs/environment-setup)
```
Comment (miklasr) [inline docs/development/installation.md]: Change to "Following these steps" or something with better grammar

**Review by miklasr — CHANGES_REQUESTED** [review]
Looks really god, nice documentation on development setup. One minor grammar fix would be nice.

**Review by miklasr — APPROVED** [review]
Awesome stuff

**Review by Ashwin-MJ — APPROVED** [review]

**Review by Andrew-Ritchie — APPROVED** [review]
Awesome work! Got it running on my phone no problem. :)

**Pull request merged**